src/pybind/mgr/progress/module.py

```diff
@@ -210,7 +210,7 @@ def global_event_update_progress(self, pg_dump, log):
             state = pg['state']
 
             states = state.split("+")
-            if pg['reported_epoch'] < self._start_epoch:
+            if int(pg['reported_epoch']) < self._start_epoch:
```

Comment:
i think we should fix the dataset used by `TestPgRecoveryEvent.test_pg_update()`. where the corresponding value of `"reported_epoch"` should be int, not str.

Reply:
also, would be better if the change can be accompanied with some analysis in the commit message.

Reply:
@tchaikov I have updated the commit message. Looks like we are casting reported_epoch in other places too, @kamoltat can we address https://github.com/ceph/ceph/pull/42034#discussion_r658884760 a follow-up PR

Reply:
@neha-ojha i meant, we need to understand where a str `pg['reported_epoch']` comes from, instead of just casting it to the type we want.

Reply:
@tchaikov it is coming from `pg_dump` dict being passed to `global_event_update_progress ` https://github.com/ceph/ceph/blob/master/src/pybind/mgr/progress/module.py#L198-L213

Reply:
@tchaikov the issue was seen in https://pulpito.ceph.com/yuriw-2021-06-23_17:06:05-rados-wip-yuri5-testing-2021-06-23-0839-pacific-distro-basic-smithi/6187045/, independent of `test_pg_update`

Reply:
> @tchaikov it is coming from the `pg_dump` dict being passed to `global_event_update_progress ` https://github.com/ceph/ceph/blob/master/src/pybind/mgr/progress/module.py#L198-L213

that's its caller, not where `pg['reported_epoch']` comes from. and it does not explain the root cause. one of the source is 

https://github.com/ceph/ceph/blob/62e3a5c41c246479c41e1360358cc2f88b99d33e/src/osd/osd_types.cc#L2823

which always dump an int. that's why i don't think the commit message explains the reason why we need this fix.

Reply:
@neha-ojha probably #42045 is a better fix.

Reply:
@tchaikov absolutely, thanks! I will repurpose this PR to remove the existing casting.